### ISSUE TITLE
text style transfer discriminator only trained on real samples?

### ISSUE THREAD

**Issue by J-Fo-S** [asyml/texar#64]
Hello, I'm having trouble understanding how your "text style transfer" model works, given that you changed it from discriminating between generated and real samples (i.e. the "discriminator" is only trained on real data). Please explain, that and the independency constraint which you removed. <del>After 6 epochs of training (approximately 18 hours), all the samples are merely duplications. How should I interpret this and how many epochs of training must I complete before I expect the samples to represent what you show in the paper?</del>

<b>Update: you can expect generated output to begin diverging around epoch 12 - for me a day and a half of training.</b>

**Comment by J-Fo-S** [comment]
I should also note that your code, for me, does not resume training from checkpoints, but starts all over if one interrupts and restarts training. I can fix this of course, but I assume you intended your code to resume training from the latest checkpoint, so fyi.

**Comment by J-Fo-S** [comment]
Update, for anyone, is that you can expect generated output to begin diverging around epoch 12 - for me a day and a half of training.

**Comment by ZhitingHu** [comment]
For `why the model would work`, please read the paper `toward controlled generation of text` For details.

For resuming from checkpoint, did you specify `—resume`?

**Comment by J-Fo-S** [comment]
Thank you, I did not specify resume. I am aware of the paper. What I am referring to actually are the changes made in this model from the model described in the paper, and how to interpret those changes. In particular, should I consider this model no longer using the generator samples on the discriminator and dropping the independency constraint as trivial, or something I should treat with greater importance when understanding this particular model in code?

**Comment by ZhitingHu** [comment]
This is a simplified implementation of the model described in the model, and its performance is reasonably strong.

- The discriminator here (and in the paper) is a sentiment classifier, rather than a real/fake classifier. In the code we only use real positive/negative data to pre-train the classifier and fix it. Additionally using generator samples to fine-tune the classifier could further improve the classification accuracy (as described in the paper), but we omitted it as the addition of classification accu is not critical for transfer quality. And there are alternative, more straightforward ways to improve the accu — you may use a better classifier architecture, e.g., the classifier based on BERT (see our [BERT example](
https://github.com/asyml/texar/tree/master/examples/bert)).

- Independency constraint is omitted just for simplicity.

Overall, the model in code implements the key idea in the paper that uses a style classifier as a direct supervision over the generated samples to drive style change.

**Comment by J-Fo-S** [comment]
Thank you very much for the explanation.
